tools/accuracy_checker/openvino/tools/accuracy_checker/evaluators/custom_evaluators/opennmt_encoder_decoder_generator_evaluator.py

```diff
@@ -0,0 +1,473 @@
+                    if decode_strategy.done:
+                        break
+                select_indices = decode_strategy.select_indices.squeeze()
+                if any_finished:
+                    self.decoder.reorder_state(select_indices, ('memory', 'mem_len'))
+                if decode_strategy.beam_size > 1 or any_finished:
+                    self.decoder.reorder_state(select_indices, ('h', 'c', 'input_feed'))
+
+            prediction = np.array(decode_strategy.predictions)[..., :-1]
+            prediction = self.adapter.process({'pred': np.transpose(prediction, [2, 1, 0])}, identifiers, None)
+            predictions.append(prediction)
+
+        return raw_outputs, predictions
+
+    def reset(self):
+        self.processing_frames_buffer = []
+        if self._encoder_predictions is not None:
+            self._encoder_predictions = []
+
+    def save_encoder_predictions(self):
+        if self._encoder_predictions is not None:
+            prediction_file = Path(self.network_info['encoder'].get('predictions', 'encoder_predictions.pickle'))
+            with prediction_file.open('wb') as file:
+                pickle.dump(self._encoder_predictions, file)
+
+    def save_decoder_predictions(self):
+        if self._decoder_predictions is not None:
+            prediction_file = Path(self.network_info['decoder'].get('predictions', 'decoder_predictions.pickle'))
+            with prediction_file.open('wb') as file:
+                pickle.dump(self._encoder_predictions, file)
+
+
+class StatefulModel:
+    state_names = []
+    state_inputs = []
+    state_outputs = []
+    state = {}
+
+    def init_state(self, *args):
+        self.state = dict(zip(self.state_names, args))
+
+    def preprocess_state(self, name, value):
+        return value
+
+    def fit_to_input(self, input_data):
+        names = input_data.keys()
+        for state_name, model_name in zip(self.state_names, self.state_inputs):
+            if state_name not in names:
+                value = self.preprocess_state(state_name, self.state[state_name])
+                input_data[model_name] = value
+        return super().fit_to_input(input_data)
+
+    def propagate_output(self, data):
+        for state_name, model_name in zip(self.state_names, self.state_outputs):
+            if model_name in data.keys():
+                self.state[state_name] = data[model_name]
+        super().propagate_output(data)
+
+
+class CommonDLSDKModel(BaseDLSDKModel):
+    default_model_suffix = 'encoder'
+    input_layers = []
+    output_layers = []
+    return_layers = []
+
+    def predict(self, identifiers, input_data, callback=None):
+        input_data = self.fit_to_input(input_data)
+        results = self.exec_network.infer(input_data)
+        self.propagate_output(results)
+        names = self.return_layers if len(self.return_layers) > 0 else self.output_layers
+        return tuple([results[name] for name in names])
+
+    def release(self):
```

Comment:
why do you need overload release method?

Reply:
Fixed